src/main/java/org/apache/maven/plugins/checkstyle/exec/DefaultCheckstyleExecutor.java

```diff
@@ -316,6 +258,73 @@ public URLClassLoader run()
         return checkerListener.getResults();
     }
 
+    private void setUpCheckstyleClassloader( Checker checker,
+                                             List<String> classPathStrings,
+                                             List<String> outputDirectories )
+        throws CheckstyleExecutorException
+    {
+        final List<URL> urls = new ArrayList<>( classPathStrings.size() );
+
+        for ( String path : classPathStrings )
+        {
+            try
+            {
+                urls.add( new File( path ).toURI().toURL() );
```

Comment:
Paths.get(path).toURI() would also work. I'm not sure if this matters in any way.